packages/dcos-integration-test/extra/test_networking.py

```diff
@@ -33,165 +37,313 @@ def lb_enabled():
+        host2 = agents[0]
+        if not self.samehost:
+            host2 = agents[1]
+        log.debug('host1 is is: {}'.format(host1))
+        log.debug('host2 is is: {}'.format(host2))
+
+        origin_app, app_uuid = self.vip_app(self.num, self.container, self.vipnet, host1, self.vip)
+        origin_app['acceptedResourceRoles'] = ['*', 'slave_public']
+        proxy_app, _ = self.vip_app(self.num, self.container, self.proxynet, host2, None)
+        proxy_app['acceptedResourceRoles'] = ['*', 'slave_public']
+
+        returned_uuid = None
+        with contextlib.ExitStack() as stack:
+            stack.enter_context(dcos_api_session.marathon.deploy_and_cleanup(origin_app, timeout=timeout))
+            sp = stack.enter_context(dcos_api_session.marathon.deploy_and_cleanup(proxy_app, timeout=timeout))
+            proxy_host = sp[0].host
+            proxy_port = sp[0].port
+            if proxy_port == 0 and sp[0].ip is not None:
+                proxy_port = backend_port_st + self.num
+                proxy_host = sp[0].ip
+            log.debug('proxy endpoints are {}'.format(sp))
+            cmd = '/opt/mesosphere/bin/curl -s -f -m 5 http://{}/test_uuid'.format(self.vipaddr)
+            returned_uuid = ensure_routable(cmd, proxy_host, proxy_port)
+            log.debug('returned_uuid is: {}'.format(returned_uuid))
+        assert returned_uuid is not None
+        assert returned_uuid['test_uuid'] == app_uuid
+        self.log('PASSED')
+
+
+class ProxyPod(Enum):
+    # NoPod = 0
+    ItsOwnPod = 1
+    SamePodAsVip = 2
+
+
+class PodTest():
+    def __init__(self, num: int, proxy_pod: ProxyPod, proxy_net: str,
+                 vip_net: str, is_named: bool, notes: str='') -> None:
+        self.num = num
+        self.proxy_pod = proxy_pod
+        self.proxy_net = proxy_net
+        self.vip_net = vip_net
+        self.is_named = is_named
+        self.notes = notes
+        self.port = vip_port_base + num
+        self.proxy_ip = '1.1.2.{}'.format(num)
+        self.proxy_vip = '{}:{}'.format(self.proxy_ip, self.port)
+        self.vip = '1.1.1.{}:{}'.format(num, self.port)
+        self.uri = self.vip
+        if self.is_named:
+            self.vip = '/vippodtest{}:{}'.format(num, self.port)
+            self.uri = 'vippodtest{}.marathon.l4lb.thisdcos.directory:{}'.format(num, self.port)
+
+    def pod_app(self):
+        proxy = None
+        app, guid = self.create_pod(self.vip_net)
+        if self.proxy_pod == ProxyPod.SamePodAsVip:
+            assert self.proxy_net == self.vip_net
+        if self.proxy_pod == ProxyPod.ItsOwnPod:
+            # recurse to create a proxy pod
+            proxy, _ = self.create_pod(self.proxy_net)
+            # delete the proxy from the vip app
+            del app['containers'][0]
+            # delete the vip from the proxy app
+            del proxy['containers'][1]
+        self.log(app)
+        self.log(proxy)
+        return app, guid, proxy
+
+    def create_pod(self, network):
```

Comment:
Can we rename this to `create_pod_app` or something that makes it clear a pod is not being launched when this function is called?